app/keepers/keepers.go

```diff
@@ -437,6 +447,31 @@ func NewAppKeeper(
 	// Must be called on PFMRouter AFTER TransferKeeper initialized
 	appKeepers.PFMRouterKeeper.SetTransferKeeper(appKeepers.TransferKeeper)
 
+	wasmConfig, err := wasm.ReadWasmConfig(appOpts)
+	if err != nil {
+		panic("error while reading wasm config: " + err.Error())
+	}
+
+	appKeepers.WasmKeeper = wasmkeeper.NewKeeper(
+		appCodec,
+		appKeepers.keys[wasmtypes.StoreKey],
+		appKeepers.AccountKeeper,
+		appKeepers.BankKeeper,
+		appKeepers.StakingKeeper,
+		distrkeeper.NewQuerier(appKeepers.DistrKeeper),
+		appKeepers.IBCKeeper.ChannelKeeper,
+		appKeepers.IBCKeeper.ChannelKeeper,
+		&appKeepers.IBCKeeper.PortKeeper,
+		appKeepers.scopedWasmKeeper,
+		appKeepers.TransferKeeper,
+		bApp.MsgServiceRouter(),
+		bApp.GRPCQueryRouter(),
+		homePath,
+		wasmConfig,
+		strings.Join(wasmapp.AllCapabilities(), ","),
+		authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+	)
+
```

Comment:
Can't add a comment on line 513 since it wasn't changed. We need to wire-up the wasm module with IBC stack, otherwise smart contracts will not be able to create channels etc. See the snippet (and figure out if we want fee in the stack):

```go
var wasmStack porttypes.IBCModule
	wasmStack = wasm.NewIBCHandler(app.WasmKeeper, app.IBCKeeper.ChannelKeeper, app.IBCFeeKeeper)
	wasmStack = ibcfee.NewIBCMiddleware(wasmStack, app.IBCFeeKeeper)
...
ibcRouter := porttypes.NewRouter().
     AddRoute(wasmtypes.ModuleName, wasmStack).
```

Reply:
resolved: 56c28cd5efc480a4af1375ed9770010ad8411b59